pmlc/dialect/tile/transforms/expand_reshape.h

```diff
@@ -0,0 +1,20 @@
+// Copyright 2020 Intel Corporation
+
+#pragma once
+
+#include "mlir/include/IR/Builder.h"
+#include "mlir/include/IR/Value.h"
+
+namespace pmlc::dialect::tile {
+
+// Flatten src to a linear tensor. Must set the proper builder insertion point
+// before this function. Return the linear tensor.
+mlir::Value flattenTensor(mlir::OpBuilder &builder, mlir::Value src);
```

Comment:
Are these useful outside of this pass? If not I'd keep them private to the completion unit.

Reply:
I intentionally expose flattenTensor and reshapeTensor in expandTensor.h. These two functions are independent and common so that they may be reused by others.